### PR TITLE
[UXIT-1716] Create Markdown Page Component

### PR DESCRIPTION
**Pull request by mirhamasala**
## 📝 Description
This PR introduces a refactor of how we handle markdown content in our application. It enables raw markdown content access, and introduces a reusable `MarkdownPage` component for consistent page layouts - ensuring that we can use our Markdown component mappings from the `MarkdownContent` component.

**Type:** Refactor

## 🛠️ Key Changes
- Enhanced Markdown loader configuration to support raw content access
- Included body export to TypeScript declarations for Markdown content
- Created reusable `MarkdownPage` component
- Updated existing Markdown pages to use the new `MarkdownPage` component

### Test Cases
1. Markdown Content Access
   - Navigate to any markdown-based page
   - Verify content renders correctly
   - Check dev tools for proper structured data

2. Page Layout Consistency
   - Check Employee Privacy Policy page
   - Check Privacy Policy page
   - Check Terms of Service page
   - Verify consistent header and content layout

### Expected Results
- All Markdown pages should render with consistent layout
- Structured data should be present in page source
- No TypeScript errors related to Markdown types

## 📋 To-Do Before Merging
- [ ] Test all existing Markdown pages
- [ ] Verify structured data format

### CODE REVIEW

File: src/app/_components/MarkdownPage.tsx [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+import type { SeoMetadata } from '@/schemas/SeoMetadataSchema'
+
+import { MarkdownContent } from '@/components/MarkdownContent'
+import { PageHeader } from '@/components/PageHeader'
+import { StructuredDataScript } from '@/components/StructuredDataScript'
+
+type MarkdownPageProps = {
+  attributes: PageAttributes
+  body: string
```
Comment (mirhamasala) [inline src/app/_components/MarkdownPage.tsx:13]: To be changed to `Parameters<typeof ReactMarkdown>[0]['children']` once #793 is merged.

**Review by barbaraperic — APPROVED** [review]
Looks good to me :) 

One question - what is the use case of having `children` as a prop in MarkdownPage?

**Comment by mirhamasala** [comment]
@barbaraperic Nothing! Accidental refactor leftover. Removed it. Thank you. 

> One question - what is the use case of having children as a prop in MarkdownPage?

File: src/app/_components/MarkdownPage.tsx [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+  body,
+  generateStructuredData,
+}: MarkdownPageProps) {
+  const { header, seo } = attributes
+
+  return (
+    <article>
+      <StructuredDataScript structuredData={generateStructuredData(seo)} />
+      <header className="mb-6">
+        <PageHeader.Title>{header.title}</PageHeader.Title>
```
Comment (CharlyMartin) [inline src/app/_components/MarkdownPage.tsx:28]: Not sure if it’s a major issue: Should we limit the page title width to match the content below?

On `/employee-privacy-policy`, the header is wider than the body content.

File: src/app/_types/markdown.d.ts [hunk 3]
```diff
@@ -1,11 +1,11 @@
...
   const react: React.ComponentType
-  export { attributes, react }
+  export { attributes, body, react }
 }
 
 declare module '*.md' {
   const attributes: import('@/types/attributesTypes').PageAttributes
   const react: React.ComponentType
-  export { attributes, react }
+  export { attributes, body, react }
```
Comment (CharlyMartin) [inline src/app/_types/markdown.d.ts:10]: We might not need this anymore, depending on what you think about [this PR](some of the work on https://github.com/FilecoinFoundationWeb/filecoin-foundation/pull/801)

Nothing to change here, just flagging.

**Review by CharlyMartin — APPROVED** [review]
Looking good Mirha, nice addition.

File: src/app/_components/MarkdownPage.tsx (hunk 2 above)
Comment (mirhamasala) [reply src/app/_components/MarkdownPage.tsx:28]: Yes! Good catch. Thank you. I added the `prose` class to `article`.

**Pull request merged**